src/map.cpp

```diff
@@ -6038,6 +6038,44 @@ bool map::sees( const tripoint &F, const tripoint &T, const int range, int &bres
     return visible;
 }
 
+int map::obstacle_coverage( const tripoint &loc1, const tripoint &loc2 )
+{
+    if( move_cost( loc2 ) > 2 && !has_flag_ter( TFLAG_FLAT, loc2 ) ) { // Can't hide if you are standing on something.
+        return 0;
+    }
+    tripoint obstaclepos;
+    int dx = loc1.x - loc2.x;
+    int dy = loc1.y - loc2.y;
+    int ax = std::abs( dx ) * 2;
+    int ay = std::abs( dy ) * 2;
+    int offset = std::min( ax, ay ) - ( std::max( ax, ay ) / 2 );
+    bresenham( loc2, loc1, offset, 0, [&obstaclepos, &loc2]( const tripoint & new_point ) {
+        if( new_point.x == loc2.x &&
+            new_point.y == loc2.y ) { // Only adjacent tile between you and enemy is checked for cover.
```

Comment:
This condition can never evaluate to true: `bresenham` won't invoke your lambda for `loc2`.

Reply:
I don't fully understand `bresenham` function.
Shouldn't it currently skip the `loc2` point and return the second point between `loc2` and `loc1`?
Is `new_point` every iterated point between `loc2` and `loc1`?

Reply:
I decided to remove that check, looks like bresenham doesn't check the beginning of the line, and starts calculating the line from the next point. In tests it worked okay.

Reply:
> do you mean bresenham doesn't check the first point, so I can just remove

Yes, that's what I had in mind. `bresenham` is a bit of an overkill here since you only need one adjacent tile, but no big deal really.